src/net_processing.cpp

```diff
@@ -3636,6 +3636,10 @@ bool PeerLogicValidation::ProcessMessages(CNode* pfrom, std::atomic<bool>& inter
     }
     CNetMessage& msg(msgs.front());
 
+    std::vector<unsigned char> tmp(msg.m_recv.data(), msg.m_recv.data() + msg.m_recv.size());
+    if (gArgs.IsArgSet("-dumpmessages"))
+        LogMessage(pfrom, msg.m_command, tmp, true);
```

Comment:
Can this logic be moved to `ReceiveMsgBytes` so that all the message dumping is contained within the net layer?

Reply:
Yes!  It's unfortunate that we have some MessageHandler code in net.cpp.  Right now all of the logging is done is done in MessageHandler, I think we should keep it that way.  It would be nice to move PushMessage to net_processing some day.

Reply:
Yes, you're right. Even though PushMessage is in net.cpp, it's always executed on the MessageHandler thread.